src/exec.js

```diff
@@ -21,6 +23,9 @@ function execSync(cmd, opts) {
     silent: common.config.silent
   }, opts);
 
+  var maxBuffer = parseInt(opts.maxBuffer, 10) || DEFAULT_MAXBUFFER_SIZE;
+  if (isNaN(maxBuffer)) maxBuffer = DEFAULT_MAXBUFFER_SIZE;
```

Comment:
Is there a way to combine these checks into one `if`? That way we only have to assign the default value once, which is probably cleaner.